### PR TITLE
Setup api

### PR DESCRIPTION
**Pull request by chudisoft**
Project requirements
    Create an API endpoint to list all posts for a user.
    Create an API endpoint to list all comments for a user's post.
    Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.
    Your API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,9 +1,19 @@
 Rails.application.routes.draw do
+  namespace :api do
+    resources :users, only: [] do
+      resources :posts, only: [] do
+        resources :comments, only: [:index, :create]
+      end
```
Comment (OyePriscilla) [inline config/routes.rb]: - [ ] Kindly ensure that the API endpoint to get post is properly configured, at the moment no route is found for api `post#index`. 🙏

File: app/controllers/api/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      post = Post.find(params[:post_id])
+      comment = post.comments.new(comment_params.merge(user: current_user))
+
+      if comment.save
+        render json: comment, status: :created
+      else
+        render json: comment.errors, status: :unprocessable_entity
+      end
+    end
+
```
Comment (OyePriscilla) [inline app/controllers/api/comments_controller.rb]: - [ ] I couldn't confirm the functionality of this method, I was unable to post comments to the api. I got the error message screenshot below. 👇 

Untitled

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @chudisoft 👏 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Build an API that is RESTful. ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/api/comments_controller.rb (hunk 2 above)
Comment (Abenezer-Tilahun) [reply app/controllers/api/comments_controller.rb]: Fixed.

File: config/routes.rb (hunk 1 above)
Comment (Abenezer-Tilahun) [reply config/routes.rb]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @chudisoft 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes. 
✔️ Good job for Implemented Blog app: add API endpoints.


## Optional suggestions 


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

**Pull request merged**